lib/server.js

```diff
@@ -228,28 +228,30 @@ class Server {
     this.httpServer.listen(port, host);
   }
 
-  message(client, data) {
-    if (Buffer.compare(EMPTY_PACKET, data) === 0) {
+  message(client, msg) {
+    if (Buffer.compare(EMPTY_PACKET, msg) === 0) {
       client.send('{}');
       return;
     }
-    const packet = metautil.jsonParse(data) || {};
-    const { id, method, type } = packet;
-    if (id && type === 'call' && method) {
-      this.rpc(client, packet);
-    } else if (id && type === 'stream') {
-      this.stream(client, packet);
-    } else {
+    const packet = metautil.jsonParse(msg) || {};
+    const parsed = protocol.deserialize(packet);
+    if (!parsed) {
       const error = new Error('Packet structure error');
       client.error(500, { error, pass: true });
     }
+    const { type, data } = parsed;
+    const handlers = {
+      call: this.rpc,
+      stream: this.stream,
+      // event ??
+    };
+    const handler = handlers[type];
+    if (!handler) throw new Error(''); // TODO - how this should be handled?
+    handler(client, data);
```

Comment:
Same as in client - naming collision. Maybe replace `msg` with `raw`? In that way, it's more descriptive (`const packet = metautil.jsonParse(raw);`) and there's no collision.

Reply:
Class `Server` doesn't extend from the `EventEmitter`.
But it can be handled like this:
```javascript
    if (!handler) {
      const error = new Error('Packet structure error');
      client.error(400, { ...data, error });
      return;
    }
```
In that way, transport will send `Packet structure error` message and if `packet` has `id` property, it will be sent as well.

Reply:
Ok